### DISCUSSION TITLE
Comparing code generation with other CPU backend

### DISCUSSION THREAD

**Discussion by vivekvpandya** [triton-lang/triton-cpu#100] [Q&A] [answered]
How this work can be compared to triron-shared and triton-mlir project as far as code generation upto LLVM IR is concerned ?

**Comment by minjang** [comment, marked as answer]
We didn't do any specific generated code comparison with triton-shared. At the end of the day, all these approaches lower TritonIR to LLVM IR. So, in that sense, it's highly similar. We still use the same LLVM backend for code generation. The job of the triton compiler is lowering to LLVM IR (sometimes it directly generates inlined assembly code, though). For a simple vector-like operation, we should see the very similar machine code. But, for more complex code like GEMM, the generated code can be different due to the different lowering.
